main/main.go

```diff
@@ -39,159 +45,204 @@ const (
 	seqNumberSetError            // 6
 	statusReportingError         // 7
 	settingsNotFoundError        // 8
+	versionMismatchError         // 9
+	jsonParsingError             // 10
 )
 
-func install() {
-	infoLogger.Println("Installed Succesfully.")
-}
-
-func enable() {
-	infoLogger.Println("Enabled Successfully.")
-}
-
-func disable() {
-	infoLogger.Println("Disabled Successfully.")
+// extension specific PublicSettings
+type extensionPublicSettings struct {
+	Name string `json:"name"`
 }
 
-func uninstall() {
-	infoLogger.Println("Uninstalled Successfully.")
+// extension specific ProtectedSettings
+type extensionPrivateSettings struct {
+	SecretString       string   `json:"secretString"`
+	SecretScript       string   `json:"secretScript"`
+	FileURLs           []string `json:"fileUris"`
+	StorageAccountName string   `json:"storageAccountName"`
+	StorageAccountKey  string   `json:"storageAccountKey"`
 }
 
-func update() {
-	infoLogger.Println("Updated Successfully.")
-}
+func install() {
+	operation := "install"
+	infoLogger.Printf("Extension MrSeq: %d, Environment MrSeq: %d", extensionMrSeq, environmentMrSeq)
+	operationLogger.Printf("[Seq Num: %d] [operation: %s]", environmentMrSeq, operation)
 
-func parseJSON(filename string) error {
-	//	Open the provided file
-	jsonFile, err := os.Open(filename)
+	err := status.ReportTransitioning(environmentMrSeq, operation, "installation in progress")
+	infoLogger.Println("Installation in progress")
 	if err != nil {
-		return errors.Wrapf(err, "Failed to open \"%s\"", filename)
+		errorLogger.Printf("Status reporting error: %+v", err)
+		os.Exit(statusReportingError)
```

Comment:
Have a central error logging mechanism instead of having a separate `Exit` for each function. In that central error logger, make sure to `status.ReportFailure` properly